content/modeling/modeling_3.md

```diff
@@ -0,0 +1,117 @@
++++
+title= "Set-up and run fractribution package"
+weight = 3
+post = ""
++++
+
+The snowplow_web_page_views (created by the snowplow_web package) and sample_events_fractribution tables will be used to demonstrate how to set-up and run the snowplow_fractribution dbt package to create the tables needed for fractional attribution.
+
+***
+
+#### **Step 1:** Set-up Variables
+
+The snowplow_fractribution dbt package comes with a list of variables specified with a default value that you may need to overwrite in your own dbt project `dbt_project.yml` file. For details you can have a look at the installed package's default variables which can be found at `[dbt_project_name]/dbt_packages/snowplow_fractribution/dbt_project.yml`.
+
+For the sake of simplicity we have selected the variables that you will most likely need to overwrite, the rest can be changed at a later stage if and when it is needed.
+
+- `conversion_window_start_date`: The start date in UTC for the window of conversions to include
+- `conversion_window_end_date`: The end date in UTC for the window of conversions to include
+- `conversion_hosts`: url_hosts to consider
+- `path_lookback_steps`: The limit for the number of marketing channels to look at before the conversion (default is 0 = unlimited)
+- `path_lookback_days`: Restrict the model to marketing channels within this many days of the conversion (values of 30, 14 or 7 are recommended)
+- `path_transforms`: An array of path transforms and their arguments (see below section **Path Transform Options**)
+- `consider_intrasession_channels`: Boolean. If false, only considers the channel at the start of the session (i.e. first page view). If true, considers multiple channels in the conversion session as well as historically.
+
+The default source schemas and tables used by the snowplow_fractribution package are:
+- *derived.snowplow_web_page_views* for the page_views data (page_views_source)
+- *atomic.events* for the Snowplow event data (conversions_source)
+
+If either of these differ in your warehouse, set the correct names as variables in your own dbt_project.yml, e.g.:
+-  `page_views_source`: `custom_schema_derived.snowplow_web_page_views`
+-  `conversions_source`: `custom_schema_derived.snowplow_ecommerce_transaction_interactions`
```

Comment:
```suggestion
-  `conversions_source`: `custom_schema_derived.conversions`
```

Reply:
Just realised the ecommerce model will only support BigQuery at first. Let's change these only once it becomes available for Snowflake, too

Reply:
Ah. So does this mean that I should change the tracking steps and all references relating to the new Ecommerce plugin back to what I had originally?

Reply:
The tracking bit for Ecommerce should all point at the "old" ecommerce plugin as the new one is not yet released (and even if it gets released in a few days unfortunately we have to wait probably till mid-January until we get Snowflake support, or till the Fractribution dbt package starts supporting BigQuery). Peter had a look and said currently the transaction API bit is from the new Ecommerce one, but the plugin is from the old one, therefore the transaction API part needs to be changed. He said he will kindly make that change (probably tomorrow).

Reply:
Yeah - I changed some references from the old one to the new when Peter recommended the new Ecommerce model. If he wants to make the changes, that's great! The TODOs I left in there in regards to the new Ecommerce plugin name uncertainties I had will need to be removed also.

Reply:
This is done now, too, Peter has revised it!